ParabolInc/parabol
# chore(teams-limit): query HubSpot API to filter out in progress deals

Fixes #7436

In this PR we query HubSpot API according to this [guide](https://www.notion.so/parabol/How-to-identify-in-progress-deals-in-HubSpot-via-API-9563a8994dce44eda7ae0c691d376a5c) to skip organizations with active deal in progress

I went with this approach as the fastest to implement, considering we are doing an experiment

I also disabled `teamLimitsCheck` function for organizations without active domains at all.

**How to test:**

- Add `HUBSPOT_API_KEY` env variable to `.env` file
- Add `HUBSPOT_SALES_PIPELINE_ID`
- Add `HUBSPOT_SALES_PIPELINE_ACTIVE_STAGES`
- Have an organization with a fake active domain (warning — do not use real customer domains)
- Reset org and add feature flag:
```
r.db('actionDevelopment').table('Organization')
  .getAll(r.args(['PUT_YOUR_ORG_ID_HERE']), {index: 'id'})
  .update({
    featureFlags: r.row('featureFlags').default([]).setInsert('teamsLimit'),
    tierLimitExceededAt: null,
    scheduledLockAt: null,
    lockedAt: null
});
```
- Uncomment https://github.com/ParabolInc/parabol/blob/99e3106cc942b8b74dea6668487fcc6658bede1e/packages/server/billing/helpers/teamLimitsCheck.ts#L14
- Trigger limits check function by running a meeting and completing it
- See that because the domain does not exists on HubSpot, the limit was successfully triggered (user see a snackbar etc)
- Now find a real deal in progress on a HubSpot, copy the domain and edit the code here: https://github.com/ParabolInc/parabol/blob/05fb6ae752918fb91faf2a10817002ab7020abcd/packages/server/billing/helpers/teamLimitsCheck.ts#L145 
- Reset the organization 
- Trigger limits check function by running a meeting and completing it
- See that no limit was triggered
- Reset the organization, remove edits to the domain (use the fake one)
- Put wrong value in `HUBSPOT_ACCESS_TOKEN` (but not empty value)
- Restart app `yarn dev`
- Trigger limits check function by running a meeting and completing it
- See that no limit was triggered
